src/NBitcoin/Consensus.cs

```diff
@@ -0,0 +1,168 @@
+﻿using System;
+using NBitcoin.BouncyCastle.Math;
+
+namespace NBitcoin
+{
+    public enum BuriedDeployments : int
+    {
+        /// <summary>
+        /// Height in coinbase
+        /// </summary>
+        BIP34,
+
+        /// <summary>
+        /// Height in OP_CLTV
+        /// </summary>
+        BIP65,
+
+        /// <summary>
+        /// Strict DER signature
+        /// </summary>
+        BIP66
+    }
+
+    public class Consensus
+    {
+        /// <summary>
+        /// An extension to <see cref="Consensus"/> to enable additional options to the consensus data.
+        /// </summary>
+        public class ConsensusOptions
+        {
+        }
+
+        public ConsensusOptions Options { get; set; }
+
+        public class BuriedDeploymentsArray
+        {
+            readonly int[] heights;
+
+            public BuriedDeploymentsArray()
+            {
+                this.heights = new int[Enum.GetValues(typeof(BuriedDeployments)).Length];
+            }
+
+            public int this[BuriedDeployments index]
```

Comment:
```
public int this[BuriedDeployments index]
{
    get => this.heights[(int)index];
    set => this.heights[(int)index] = value;
}
```

probably looks nicer (but I don't have a strong point here so ignore if you like it how it is)

same for line 60 and 106

Reply:
💎

Reply:
🏄